earn/src/components/borrow/modal/tab/AddUniswapNFTAsCollateralTab.tsx

```diff
@@ -0,0 +1,381 @@
+      setPendingTxn(contractData);
+      setIsPending(false);
+      setIsOpen(false);
+    } else if (!contractIsLoading && !contractDidSucceed) {
+      setIsPending(false);
+    }
+  }, [contractDidSucceed, contractData, contractIsLoading, setPendingTxn, setIsOpen]);
+
+  let confirmButtonState = ConfirmButtonState.READY;
+
+  if (approvingTxn !== null) {
+    confirmButtonState = ConfirmButtonState.APPROVING;
+  } else if (isPending) {
+    confirmButtonState = ConfirmButtonState.PENDING;
+  } else if (getApprovedData !== ALOE_II_UNISWAP_NFT_MANAGER_ADDRESS) {
+    confirmButtonState = ConfirmButtonState.APPROVE_NFT_MANAGER;
+  } else if (contractWriteConfig && contractWriteConfig.request === undefined) {
+    confirmButtonState = ConfirmButtonState.LOADING;
+  }
+
+  const confirmButton = getConfirmButton(confirmButtonState);
+
+  return (
+    <FilledStylizedButton
+      size='M'
+      fillWidth={true}
+      disabled={!confirmButton.enabled}
+      onClick={() => {
+        if (confirmButtonState === ConfirmButtonState.APPROVE_NFT_MANAGER) {
+          setIsPending(true);
+          writeApproveAsync?.({
+            recklesslySetUnpreparedArgs: [
+              ALOE_II_UNISWAP_NFT_MANAGER_ADDRESS,
+              BigNumber.from(uniswapNFTPosition[0].toFixed(0)),
+            ],
+            recklesslySetUnpreparedOverrides: { gasLimit: BigNumber.from(100000) },
+          })
+            .then((txnResult) => {
+              setApprovingTxn(txnResult);
+              txnResult
+                .wait(1)
+                .then(() => {
+                  refetchGetApprovedData();
+                })
+                .finally(() => {
+                  setApprovingTxn(null);
+                  setIsPending(false);
+                });
+            })
+            .catch((_err) => {
+              setApprovingTxn(null);
+              setIsPending(false);
+            });
+        } else if (confirmButtonState === ConfirmButtonState.READY) {
+          setIsPending(true);
+          contractWrite?.();
+        }
+      }}
+    >
+      {confirmButton.text}
+    </FilledStylizedButton>
+  );
+}
+
+export type AddUniswapNFTAsCollateralTabProps = {
+  marginAccount: MarginAccount;
+  uniswapNFTPositions: Map<number, UniswapNFTPosition>;
+  defaultUniswapNFTPosition: [number, UniswapNFTPosition];
+  setIsOpen: (open: boolean) => void;
+  setPendingTxn: (pendingTxn: SendTransactionResult | null) => void;
+};
+
+export function AddUniswapNFTAsCollateralTab(props: AddUniswapNFTAsCollateralTabProps) {
+  const { marginAccount, uniswapNFTPositions, defaultUniswapNFTPosition, setIsOpen, setPendingTxn } = props;
+  const [currentPage, setCurrentPage] = useState(1);
+  const [selectedTokenId, setSelectedTokenId] = useState<number>(defaultUniswapNFTPosition[0]);
+
+  const { address: userAddress } = useAccount();
+
+  const filteredPages: [number, UniswapNFTPosition][][] = useMemo(() => {
+    const pages: [number, UniswapNFTPosition][][] = [];
```

Comment:
Data types getting a little complicated. Maybe define type for `[number, UniswapNFTPosition]`?